consensus/hotstuff/safetyrules/safety_rules.go

```diff
@@ -13,16 +13,16 @@ type SafetyRules struct {
 	signer        hotstuff.Signer
 	forks         hotstuff.ForksReader
 	persist       hotstuff.Persister
-	committee     hotstuff.Committee // only produce votes when we are valid committee members
-	lastVotedView uint64             // need to keep track of the last view we voted for so we don't double vote accidentally
+	committee     hotstuff.Replicas // only produce votes when we are valid committee members
```

Comment:
@AlexHentschel I think we need to use dynamic committee here, replica always votes for block and we don't want ejected nodes to vote on block. No need to fix this, I just want to double check this.

Reply:
Thanks for noticing this @durkmurder. I would agree that we need to use the dynamic committee here. 
Though, I have a slightly different view on the reasoning for why it should be the dynamic committee:
* For an ejected node, it is of little concern whether or not it votes. Especially if the node is byzantine, why wouldn't it vote? Of course we desire honest nodes not to vote once we enter the epoch where they are unstaking. But I think this would also be achieved by the static committee.  
* I think we need the dynamic committee on the recipients' end:
  * An honest replica should not vote for a block from an ejected node. Hence, the dynamic committee. 
  * Ideally, an honest replica should also exclude votes from ejected replicas.

Reply:
Copying this question here: https://github.com/onflow/flow-go/pull/2328#discussion_r867209653 regarding which of the these statements we care about more 😅 

> For an ejected node, it is of little concern whether or not it votes

> an honest replica should also exclude votes from ejected replicas.